src/components/Planning/GroupForm/GroupForm.tsx

```diff
@@ -0,0 +1,333 @@
+
+  const [formState, setFormState] = useState<IFormState>({
+    isOpen: false,
+    selectedClass: '',
+    projectsForSubmit: [],
+    selectedLocation: '',
+    showAdvanceFields: false,
+  });
+
+  const { isOpen, selectedClass, selectedLocation, projectsForSubmit, showAdvanceFields } =
+    formState;
+
+  useClassList(true, selectedClass);
+  useLocationList(true, selectedLocation);
+
+  useEffect(() => {
+    const subscription = watch((value, { name }) => {
+      if (name === 'masterClass' && value.masterClass?.value) {
+        setFormState((current) => ({ ...current, selectedClass: value.masterClass?.value }));
+        setValue('class', { label: '', value: '' });
+        setValue('subClass', { label: '', value: '' });
+      } else if (name === 'class' && value.class?.value) {
+        setFormState((current) => ({ ...current, selectedClass: value.class?.value }));
+        setValue('subClass', { label: '', value: '' });
+      } else if (name === 'subClass' && value.subClass?.value) {
+        setFormState((current) => ({ ...current, selectedClass: value.subClass?.value }));
+        if (!value.class?.value || !value.masterClass?.value) {
+          const { _class, subClass, masterClass } = getReverseClassHierarchy(value.subClass?.value);
+          setValue('masterClass', masterClass);
+          setValue('class', _class);
+          setValue('subClass', subClass);
+        }
+      }
+
+      if (name === 'district' && value.district?.value) {
+        setFormState((current) => ({ ...current, selectedLocation: value.district?.value }));
+        setValue('division', { label: '', value: '' });
+        setValue('subDivision', { label: '', value: '' });
+      } else if (name === 'division' && value.division?.value) {
+        setFormState((current) => ({ ...current, selectedLocation: value.division?.value }));
+        setValue('subDivision', { label: '', value: '' });
+      } else if (name === 'subDivision' && value.subDivision?.value) {
+        setFormState((current) => ({ ...current, selectedLocation: value.subDivision?.value }));
+        if (!value.division?.value || !value.district?.value) {
+          const { division, subDivision, district } = getReverseLocationHierarchy(
+            value.subDivision?.value,
+          );
+          setValue('district', district);
+          setValue('division', division);
+          setValue('subDivision', subDivision);
+        }
+      }
+    });
+    return () => subscription.unsubscribe();
+  }, [watch, setValue, getReverseClassHierarchy, getReverseLocationHierarchy]);
+
+  const onSubmit = useCallback(
+    async (form: IGroupForm) => {
+      console.log(buildRequestPayload(form, projectsForSubmit));
+      dispatch(postGroupThunk(buildRequestPayload(form, projectsForSubmit))).then(() => {
+        reset(formValues);
+        setFormState((current) => ({
+          ...current,
+          showAdvanceFields: false,
+          projectsForSubmit: [],
```

Comment:
Could projectsForSubmit be a part of the form?

Reply:
Added as part of form hook